Adamant/Modules/PartnerQR/PartnerQRViewModel.swift

```diff
@@ -53,7 +53,20 @@ final class PartnerQRViewModel: NSObject, ObservableObject {
     }
     
     func renameContact() {
-        guard let alert = self.makeRenameAlert(for: self.title) else { return }
+        let alert = AlertFactory.makeRenameAlert(
+            titleFormat: String(format: .adamant.chat.actionsBody, self.title),
+            placeholder: .adamant.chat.name,
+            initialText: self.addressBookService.getName(for: self.title) ?? self.partnerName
+        ) { [weak self] newName in
+            guard let self = self else { return }
+            
+            Task {
```

Comment:
should we add @MainActor here? Can there be any potential problems with concurrency code?